tests/contrib/celery/test_tagging.py

```diff
@@ -0,0 +1,141 @@
+import socket
+import time
+
+from celery import Celery
+from celery.contrib.testing.worker import start_worker
+import pytest
+
+from ddtrace import Pin
+from ddtrace.contrib.celery import patch
+from ddtrace.contrib.celery import unpatch
+from tests.utils import DummyTracer
+
+from .base import AMQP_BROKER_URL
+from .base import BACKEND_URL
+from .base import BROKER_URL
+
+
+redis_celery_app = Celery(
+    "mul_celery",
+    broker=BROKER_URL,
+    backend=BACKEND_URL,
+)
+
+
+@redis_celery_app.task
+def multiply(x, y):
+    return x * y
+
+
+amqp_celery_app = Celery(
+    "add_celery",
+    broker=AMQP_BROKER_URL,
+    backend="rpc://",
+)
+
+
+@amqp_celery_app.task
+def add(x, y):
+    return x + y
+
+
+@pytest.fixture(autouse=False)
+def instrument_celery():
+    # Instrument Celery and create an app with Broker and Result backends
+    patch()
+    yield
+    # Remove instrumentation from Celery
+    unpatch()
+
+
+@pytest.fixture(scope="session")
+def celery_config():
+    return {"broker_url": BROKER_URL}
+
+
+@pytest.fixture
+def dummy_tracer():
+    return DummyTracer()
+
+
+@pytest.fixture(autouse=False)
+def traced_redis_celery_app(instrument_celery, dummy_tracer):
+    Pin.get_from(redis_celery_app)
+    Pin.override(redis_celery_app, tracer=dummy_tracer)
+    yield redis_celery_app
+
+
+@pytest.fixture(autouse=False)
+def traced_amqp_celery_app(instrument_celery, dummy_tracer):
+    Pin.get_from(amqp_celery_app)
+    Pin.override(amqp_celery_app, tracer=dummy_tracer)
+    yield amqp_celery_app
+
+
+def test_redis_task(traced_redis_celery_app):
+    tracer = Pin.get_from(traced_redis_celery_app).tracer
+
+    with start_worker(  # <-- Important!
+        traced_redis_celery_app,
+        pool="solo",
+        loglevel="info",
+        perform_ping_check=False,
+        shutdown_timeout=30,  # <-- Important!
+    ):
+        t = multiply.delay(4, 4)
+        assert t.get(timeout=2) == 16
+
+        # wait for spans to be received
+        time.sleep(3)
+
+        assert_traces(tracer, "multiply", t, "redis")
+
+
+def test_amqp_task(instrument_celery, traced_amqp_celery_app):
+    tracer = Pin.get_from(traced_amqp_celery_app).tracer
+
+    with start_worker(  # <-- Important!
+        traced_amqp_celery_app,
+        pool="solo",
+        loglevel="info",
+        perform_ping_check=False,
+        shutdown_timeout=30,  # <-- Important!
+    ):
+        t = add.delay(4, 4)
+        assert t.get(timeout=2) == 8
+
+        # wait for spans to be received
+        time.sleep(3)
+
+        assert_traces(tracer, "add", t, "amqp")
+
+
+def assert_traces(tracer, task_name, task, backend):
```

Comment:
Should we just make this into a snapshot test?

Reply:
oh good idea!